pkgs/development/python-modules/cylp/default.nix

```diff
@@ -0,0 +1,73 @@
+{ lib
+, buildPythonPackage
+, fetchFromGitHub
+, numpy
+, scipy
+, python
+, pytestCheckHook
+, cython
+, cbc
+, bzip2
+, zlib
+, pkg-config
+}:
+
+buildPythonPackage rec {
+  pname = "cylp";
+  version = "0.91.4";
+
+  src = fetchFromGitHub {
+    owner = "coin-or";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "sha256-m09s+QrkLGtF83tDLNjU4ECOqgW5RT0h9+fSwDeADag=";
+  };
+
+  nativeBuildInputs = [
+    pkg-config
+    cython
+  ];
+
+  buildInputs = [
+    cbc
+    bzip2
+    zlib
+  ];
+
+  propagatedBuildInputs = [
+    numpy
+    scipy
+  ];
+
+  preBuild = ''
+    # Leak some memory, reverting this commit. Otherwise there's a segfault
+    # that's being debugged in https://github.com/coin-or/CyLP/issues/105
+    substituteInPlace cylp/cy/CyClpSimplex.pyx --replace \
+      'del self.CppSelf' \
+      'pass'
+
+    # Rebuild all cython files
+    export CYLP_USE_CYTHON=1
+  '';
+
+  checkInputs = [ pytestCheckHook ];
+  pythonImportsCheck = [ "cylp" ];
+  pytestFlagsArray = [ "-s" ];
+
+  preCheck = ''
+    cd cylp/tests
+    rm __init__.py
+  '';
+
+  # https://github.com/coin-or/CyLP/commit/e5bab3cca81c0ad9886bcd8127923a436bcdaf9b
+  disabledTestPaths = [
+    "test_MIP.py"
```

Comment:
It makes me feel a little uncomfortable disabling a test that is core functionality: from the description, "A Python interface to CLP, CBC, and CGL to solve LPs and MIPs"

Reply:
I agree, but unfortunately this particular test is _obviously_ broken because upstream removed a method without removing the calls to that method from inside the tests. I have verified that the functionality does work. Do you want me to add one of my own tests?

Reply:
I should have left more comments related to that URL, but the URL is relevant because this commit removed the branchAndBound function, and every test in that files calls the branchAndBound function.

Thus, unfortunately this particular test is _obviously_ broken because upstream due to the missing. I have verified that the functionality for solving MIPS does in fact work. Do you want me to add one of my own tests?

Reply:
Was the upstream warned about this lack of updates?

You do not need to run your own tests here, in this particular case it is the upstream that is in error.

Reply:
Yes, the upstream is definitely aware. I may submit a patch to fix these tests to upstream when I have some free time as well.

Reply:
Looks like upstream was just reminded of the issue in coin-or/CyLP#168.
It doesn't feel like they're going to fix this anytime soon unless someone provides them a new stimulus of some sort.